tests/unit-tests/test-class-usage-tracking.php

```diff
@@ -377,14 +404,27 @@ private function setupAjaxRequest() {
 	 * Helper method to set up tracking opt-in dialog.
 	 */
 	private function setupOptInDialog() {
-		// Set manage_sensei cap on current user
-		$user = wp_get_current_user();
-		$user->add_cap( 'manage_sensei' );
+		// Ensure current user can enable tracking
+		$this->allowCurrentUserToEnableTracking();
 
 		// Ensure setting is not set
-		Sensei()->settings->set( self::PREFIX . '_usage_tracking_enabled', false );
+		$this->setTrackingEnabled( false );
 
-		// Reset the in-memory settings
-		Sensei()->settings->get_settings();
+		// Refresh flag
+		$this->refreshTrackingEnabledFlag();
+	}
+
+	/**
+	 * Helper method to set tracking enabled or disabled. Note: this calls the
+	 * private `set_tracking_enabled` on the {Prefix}_Usage_Tracking class.
+	 *
+	 * @param bool $enable true if tracking should be enabled, false otherwise.
+	 * Default: true.
+	 **/
+	private function setTrackingEnabled( $enable = true ) {
+		$reflection = new \ReflectionClass( get_class( $this->usage_tracking ) );
```

Comment:
I did this to make the tests less Sensei-specific. In particular, this function is already implemented in the Usage Tracking class, so I didn't want to have to duplicate the implementation for the tests. But I agree, it's a bit hacky.

This is something that the abstract superclass idea (from the PR description) would actually help a lot. The tests could use their own subclass, and not be dependent on the plugin-specific implementation at all. I'm starting to think that it might be pretty beneficial to change over to that architecture...